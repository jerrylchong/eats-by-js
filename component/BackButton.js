import React from 'react';
import {Image, TouchableOpacity, StyleSheet, View, Dimensions} from "react-native";

const BackButton = (props) => {
    return (
        <View style = {props.style}>
            <TouchableOpacity style = {styles.container} onPress = {props.onPress}>
                {props.white
                    ? <Image style = {styles.image} source ={require('../assets/whiteback.png')} />
                    : <Image style = {styles.image} source ={require('../assets/backbutton.png')} />
                }
            </TouchableOpacity>
        </View>
    )
}

export default BackButton

const windowWidth= Dimensions.get('window').width;

const styles = StyleSheet.create({
    container: {
        height: windowWidth * 0.1,
        width: windowWidth * 0.1,
        alignItems: 'flex-start',
        justifyContent: 'flex-start'
    },
    image: {
        height: 13,
        width: 15
    }
})