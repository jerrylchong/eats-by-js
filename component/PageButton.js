import React from "react";
import {Text, TouchableOpacity, StyleSheet, View, Dimensions} from "react-native";

const PageButton = (prop) => {
    return (
        <View style = {styles.shadow}>
            <TouchableOpacity style = {styles.button} onPress = {prop.onPress}>
                <Text style = {{color: 'white', fontFamily: 'Ubuntu'}}>{prop.text}</Text>
            </TouchableOpacity>
        </View>
    )
}

export default PageButton

const styles = StyleSheet.create({
    shadow: {
        width: Dimensions.get('window').width * 0.8,
        height: Dimensions.get('window').width * 0.15,
        backgroundColor: '#ff6961',
        borderRadius: Dimensions.get('window').width * 0.4,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
        margin: 10
    },
    button: {
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    }
})