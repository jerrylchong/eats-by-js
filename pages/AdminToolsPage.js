import React, {useEffect} from 'react';
import {
    View,
    StyleSheet,
    Text,
    ImageBackground,
    Dimensions,
    BackHandler,
    Platform
} from "react-native";
import PageButton from "../component/PageButton";
import BackButton from "../component/BackButton";
import {useSafeArea} from "react-native-safe-area-context";

function AdminToolsPage({ navigation, route }) {

    useEffect(() => {
        const backAction = () => {
            navigation.goBack();
            return true;
        };

        const backHandler = BackHandler.addEventListener(
            "hardwareBackPress",
            backAction
        );

        return () => backHandler.remove();
    }, []);

    const insets = useSafeArea();

    return (
        <View style = {[
            styles.container,
            {paddingTop: insets.top, paddingLeft: insets.left, paddingRight: insets.right}
        ]}>
            <ImageBackground style = {styles.background} source={require('../assets/background.png')}/>
            <Text style = {styles.header}>Admin Tools</Text>
            <BackButton white={false} style={styles.back} onPress={navigation.goBack}/>
            <View style = {styles.buttons}>
                <PageButton text = "Add a Store" onPress = {() => navigation.navigate('Add')}/>
                <PageButton text = "Edit a Store" onPress = {() => navigation.navigate('Edit')}/>
                <PageButton text = "Delete a Store" onPress = {() => navigation.navigate('Delete')}/>
                <PageButton text = "View Requested Stores" onPress = {() => navigation.navigate('Requested')}/>
            </View>
        </View>
    );
}

export default AdminToolsPage

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        alignItems: 'center',
    },
    background: {
        position:'absolute',
        bottom: 0,
        width: '100%',
        height: Dimensions.get('window').width * 728/1668
    },
    back: {
        position: 'absolute',
        top: Platform.OS == "ios" ? '5%' :'2%',
        left: '3%'
    },
    header: {
        position: 'relative',
        top: '5%',
        fontSize: 20,
        marginTop: '5%',
        fontFamily: 'Ubuntu-Medium',
        color: '#404040'
    },
    buttons: {
        position: 'relative',
        top: '22%',
    }
})
