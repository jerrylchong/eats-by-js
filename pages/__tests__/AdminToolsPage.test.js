import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AdminToolsPage from "../AdminToolsPage";
import BackButton from "../../component/BackButton";
import PageButton from "../../component/PageButton";
import {SafeAreaProvider} from "react-native-safe-area-context";
import {ImageBackground, Text, View} from "react-native";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(
        <SafeAreaProvider initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
            <AdminToolsPage
                navigation={{navigate: jest.fn()}}
            />
        </SafeAreaProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });

test('test component rendering', () => {
    const wrapper = shallow(
        <SafeAreaProvider initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
            <AdminToolsPage
                navigation={{navigate: jest.fn()}}
            />
        </SafeAreaProvider>
    )
    expect(wrapper.find(ImageBackground)).toBeTruthy(); // checks if imagebackground is rendered
    expect(wrapper.find(Text)).toBeTruthy(); // checks if title text is rendered
    expect(wrapper.find(BackButton)).toBeTruthy(); // checks if backbutton is rendered
    expect(wrapper.find(View).find(PageButton).at(0)).toBeTruthy(); // checks if add store pagebutton is rendered
    expect(wrapper.find(View).find(PageButton).at(1)).toBeTruthy(); // checks if edit store pagebutton is rendered
    expect(wrapper.find(View).find(PageButton).at(2)).toBeTruthy(); // checks if delete store pagebutton is rendered
})