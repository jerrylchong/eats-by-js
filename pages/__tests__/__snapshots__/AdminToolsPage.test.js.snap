// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "column",
        },
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "bottom": 0,
          "height": 327.33812949640287,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 327.33812949640287,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 20,
          "marginTop": "5%",
          "position": "relative",
          "top": "5%",
        }
      }
    >
      Admin Tools
    </Text>
    <View
      style={
        Object {
          "left": "2%",
          "position": "absolute",
          "top": "5%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 13,
              "width": 15,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
          "top": "27%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ff6961",
            "borderRadius": 300,
            "elevation": 5,
            "height": 112.5,
            "margin": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu",
              }
            }
          >
            Add a Store
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ff6961",
            "borderRadius": 300,
            "elevation": 5,
            "height": 112.5,
            "margin": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu",
              }
            }
          >
            Edit a Store
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ff6961",
            "borderRadius": 300,
            "elevation": 5,
            "height": 112.5,
            "margin": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu",
              }
            }
          >
            Delete a Store
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
